### ISSUE TITLE
Define application state model

### ISSUE THREAD

**Issue by kadamwhite** [bocoup/image-annotator#7]
Define the data model for the application state and structure it using Redux

Depends on #4

**Comment by kadamwhite** [comment]
#9 begins this work

**Comment by kadamwhite** [comment]
Superseded by #16; we have completed the initial definition of our state model, and now #16 tracks changing the idioms we use to structure it
